### PR TITLE
KAFKA-16577: New consumer fails with stop within allotted timeout in consumer_test.py system test

### PR DESCRIPTION
**Pull request by kirktrue**
The `KafkaAsyncConsumer` would occasionally fail to stop when `wakeup()` was invoked. It turns out that there's a race condition between the thread that invokes `wakeup()` and the thread that is performing an action on the consumer. If the operation's `Future` is *already* completed by thread A when thread B invoke's `completeExceptionally()` inside `wakeup()`, the `WakeupException` will be ignored. We should use the return value from `completeExceptionally()` to determine if that call actually triggered completion of the `Future`. If that method returns `false`, that signals that the `Future` was already completed, and the exception we passed to `completeExceptionally()` was ignored. Therefore, we then need to return a new `WakeupFuture` instead of `null` so that the *next* call to `setActiveTask()` will throw the `WakeupException`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by kirktrue** [comment]
@lucasbru/@cadonna —please review. Thanks!

**Comment by kirktrue** [comment]
@cadonna—can you review this PR? Thanks!

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java [hunk 1]
```diff
@@ -43,8 +43,12 @@ public void wakeup() {
                 return new WakeupFuture();
             } else if (task instanceof ActiveFuture) {
                 ActiveFuture active = (ActiveFuture) task;
-                active.future().completeExceptionally(new WakeupException());
-                return null;
+                boolean wasTriggered = active.future().completeExceptionally(new WakeupException());
```
Comment (lucasbru) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java:46]: Could we add a test in `WakeupTriggerTest` for this?

**Review by lucasbru — COMMENTED** [review]
Thanks for the PR! Mostly looking good to me, but would be good to add a little unit test to stay consistent.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java (hunk 1 above)
Comment (kirktrue) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java:46]: Will do.

Comment (kirktrue) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java:46]: I added some unit tests.

**Comment by kirktrue** [comment]
@lucasbru—please re-review. Thanks!

**Review by lucasbru — APPROVED** [review]
LGTM, thanks!

**Pull request merged**